webrender/src/frame_builder.rs

```diff
@@ -882,12 +882,17 @@ impl FrameBuilder {
                     BoxShadowClipMode::Inset => 1.0,
                 };
 
-                // If we have a border radius, we'll need to apply
-                // a clip-out mask.
+                // Outset box shadows with border radius
+                // need a clip out of the center box.
+                let extra_clip_mode = match clip_mode {
```

Comment:
I suspect you mean clip_mode != BoxShadowClipMode::Inset? This didn't work, the border-radius clip was ignored completely and we'd just get a square inset box shadow. I suspect because the clip_mode we're passing in when we create the display item doesn't include the proper rounded rect clip. We could alter the passed in clip region to translate it to a complex one since it's passed directly by users of the API, but that is confusing to API users.